denizen_scripts/survival/repo-link/items/tiered_potions.dsc

```diff
@@ -0,0 +1,266 @@
+# -- Tiered Potions data script
+tiered_potions_data:
+  type: data
+  tiered_potion_2:
+    item: brown_mushroom
+    uses: 3
+    heals: 10
+  tiered_potion_3:
+    item: glowstone_dust
+    uses: 5
+    heals: 15
+  tiered_potion_4:
+    item: fire_charge
+    uses: 8
+    heals: 20
+  tiered_potion_5:
+    item: redstone_block
+    uses: 12
+    heals: 25
+  tiered_potion_6:
+    item: ancient_debris
+    uses: 18
+    heals: 30
+  tiered_potion_7:
+    item: nether_star
+    uses: 25
+    heals: 45
+  tiered_potion_8:
+    item: diamond_block
+    uses: 35
+    heals: 70
+  tiered_potion_9:
+    item: netherite_ingot
+    uses: 40
+    heals: 90
+  tiered_potion_10:
+    item: glistering_melon_slice
+    uses: 50
+    heals: 512
+
+# -- Tiered Potions world events
+tiered_potions_events:
+  type: world
+  debug: false
+  events:
+    # -- When the player crafts a tiered potion, the items don't have any NBT data.
+    # -- This is because `<item[grass_block]> != <item[grass_block].with[nbt=<list[has/nbt]>]>`.
+    # -- We fallback to `1` when adding one to the defined item's NBT.
+    on player consumes tiered_potion_*:
+      # Heal and add one to the usage counter.
+      - determine passively cancelled
+      - define item <context.item>
+      - heal <script[tiered_potions_data].data_key[<context.item.script.name>.heals].mul[2]>
+      - adjust <[item]> nbt:<list[uses/<[item].nbt[uses].add[1]||1>]>
+      - if <[item].nbt[uses]> >= <script[tiered_potions_data].data_key[<context.item.script.name>.uses]>:
+        # -- If `- determine <[item]>` doesn't work, try `- determine <context.item>` or `- determine cancelled:false`.
+        - playsound <player> sound:ENTITY_ITEM_BREAK
+        - determine <[item]>
+      # Wait one tick, then update the potion's NBT.
+      - wait 1t
+      - inventory adjust slot:<player.held_item_slot> nbt:<list[uses/<[item].nbt[uses]>]>
```

Comment:
This will be a notable piece to fix later on, verifying which hand this is being done from.
```suggestion
      - if <player.item_in_hand> == <context.item>:
        - inventory adjust slot:<player.held_item_slot> nbt:<list[uses/<[item].nbt[uses]>]>
      - else:
        - inventory adjust slot:<player.item_in_offhand> nbt:<list[uses/<[item].nbt[uses]>]>
```